### PR TITLE
311 Add a database library

### PR DESCRIPTION
**Pull request by ZQUEMA**
## Purpose:
This pull request will implement a light SQLite database library.

### CODE REVIEW

File: libs/Database/Database.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+** EPITECH PROJECT, 2022
+** R-Type
+** File description:
+** Database
+*/
+
+#include "Database.hpp"
+#include <iostream>
+
+Database::Database(const std::filesystem::path &fileName) : _db(NULL), _fileName(fileName) { _open(_fileName); }
```
Comment (LucasTesnier) [inline libs/Database/Database.cpp:11]: use nullptr instead of NULL

File: libs/Database/Database.hpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+** R-Type
+** File description:
+** Database
+*/
+
+#pragma once
+
+extern "C"
+{
+#include <sqlite3.h>
```
Comment (LucasTesnier) [inline libs/Database/Database.hpp:12]: missing indentation

File: libs/Database/Database.hpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    /// @param setValues The values you want to change.
+    /// @param condition The condition to be met for the update to occur.
+    /// @return The return value is a boolean value.
+    bool updateUsers(const std::string &setValue, const std::string &condition);
+
+  private:
+  protected:
+  private:
+    /// @brief the database
+    sqlite3 *_db;
```
Comment (LucasTesnier) [inline libs/Database/Database.hpp:68]: Shared / unique ptr ?

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Some minor changes

File: libs/Database/Database.hpp (hunk 2 above)
Comment (ZQUEMA) [reply libs/Database/Database.hpp:12]: It's not clang format

File: libs/Database/Database.hpp (hunk 3 above)
Comment (ZQUEMA) [reply libs/Database/Database.hpp:68]: This is a C encapsulation , it works differently and sometimes it causes problems.

**Review by LucasTesnier — DISMISSED** [review]
Some minor changes

**Review by LucasTesnier — APPROVED** [review]

**Review by TomDUVAL-MAHE — APPROVED** [review]

**Pull request merged**